ntrans.py

```diff
@@ -20,10 +20,12 @@ def __init__(
         user_choices: Dict[str, Any],
         progress_queue: queue.Queue[int],
         cancel_thread_event: threading.Event,
+        generation_finished_event: threading.Event,
     ) -> None:
         self.user_choices = user_choices
         self.progress_queue = progress_queue
         self.cancel_thread = cancel_thread_event
+        self.generation_finished_event = generation_finished_event
```

Comment:
This event is never used. You don't need it, because you use `is_alive()` instead.